### PR TITLE
Change static header

### PR DESCRIPTION
**Pull request by Saverio976**
try to optimize compilation time with:
- split function to different header (a .c will now not copy alllllll the func proto)
- pass static const that were .h to a .c and a .h that refer to them as extern

### CODE REVIEW

File: src/player/update/update_player.c [hunk 1]
```diff
@@ -23,6 +24,7 @@ static void (*update_ptr[])(player_t *, scene_t *, window_t *, float) = {
     update_roulade
 };
 
+// TODO: here why ?
```
Comment (KitetsuK) [inline src/player/update/update_player.c:27]: What's this ?

File: src/player/update/update_roulade.c [hunk 2]
```diff
@@ -30,7 +30,7 @@ static void apply_rect(player_t *player, int current_id)
     player->obj->bigdata.sprite_bigdata.rect = (sfIntRect) {rect[0], rect[1],
         rect[2], rect[3]};
     sfSprite_setOrigin(player->obj->drawable.sprite,
-        (sfVector2f) {rect[2] / 2, rect[3] / 2});
+        (sfVector2f) {rect[2] / 2.0, rect[3] / 2.0});
```
Comment (KitetsuK) [inline src/player/update/update_roulade.c:33]: Magic value ?

Comment (Saverio976) [reply src/player/update/update_roulade.c:33]: 2.0 ?
it is just that we have the width and heigh and to have the middle you divide by 2

File: src/player/update/update_player.c (hunk 1 above)
Comment (Saverio976) [reply src/player/update/update_player.c:27]: ah yeah, just a thought, because the function is empty so i was thinking of this function

**Pull request merged**